src/tactic/find.lean

```diff
@@ -54,19 +54,22 @@ do (p_pis, p) ← p.get_uninst_pis,
    match_exact p e,
    match_hyps p_pis [] pis
 
+private meta def trace_match (pat : pexpr) (ty : expr) (n : name) : tactic unit :=
+(do guard ¬ n.is_internal,
+   match_sig pat ty,
+   ty ← pp ty,
+   trace format!"{n}: {ty}") <|> skip
```

Comment:
Instead of `(do ...) <|> skip` why not `try_core $ do ...`.

Reply:
`try` is even cleaner.